src/core/workaround.rs

```diff
@@ -0,0 +1,103 @@
+/// # Workaround
+///
+/// Provides workarounds to circumvent some limitations of Rust and/or
+/// dependencies we use.
+///
+/// Currently libbpf_rs does not implement Send for many of its objects, while
+/// for some of them it is actually safe. In the meantime, SendWrapper objects
+/// can be used to send a non-Send libbpf_rs object to another thread.
+///
+/// For now we support SendWrapperMap.
+
+/// SendWrapper<T> should not be used directly but only specific typedefs ones.
+/// When adding a new SendWrapper type research should be done to ensure this is
+/// actually safe!
+///
+/// SendWrapper<T> only implements the Send trait, if Sync is needed and more
+/// than one user wants access use the common Arc<Mutex<SendWrapper<T>>>
+/// construction (see core::workaround::tests::sync).
+pub(crate) struct SendWrapper<T>(T)
```

Comment:
How about a sealed trait? I saw this somewhere and found it an interesting solution, basically the idea is to create an internal non-public module:

```rust
mod sealed {
    pub(super) trait Hacked {}
    /* Do not add anything else here unless it's absolutely necessary!!!! */
    impl Hacked for libbpf-rs::Map {}
    impl Hacked for libbpf-rs::RingBuf {}
}

/* Only a handful of hacked structs can implement this trait. And for someone to do it, they have to touch a very specific part of the code that raises the reviewer's alarms. */
pub(crate) struct SendWrapper<T: mod::Hacked>(T)
```
I don't know if this would work in this case but it might be worth a try.